app/services/name_splitter.rb

```diff
@@ -0,0 +1,32 @@
+class NameSplitter
+  def initialize(full_name:)
+    @full_name = full_name
+  end
+
+  def first_name
+    @first_name || find_first_name
+  end
+
+  def last_name
+    @last_name || find_last_name
+  end
+
+  private
+  def find_first_name
+    split_name = @full_name.split
```

Comment:
In the name of DRYness, I'd move `@split_name = @full_name.split` to `initialize`.